components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/SAMLLogoutHandler.java

```diff
@@ -94,14 +95,22 @@ public String getName() {
     protected String getSamlSSOTokenIdFromEvent(Event event) {
 
         String samlssoTokenId = null;
-        if (event.getEventProperties().get(EventProperty.REQUEST) instanceof HttpServletRequest) {
+        if (event.getEventProperties().get(EventProperty.REQUEST) != null &&
+                event.getEventProperties().get(EventProperty.REQUEST) instanceof HttpServletRequest) {
             HttpServletRequest request = (HttpServletRequest) event.getEventProperties().get(EventProperty.REQUEST);
-            if (request != null) {
                 Cookie cookie = FrameworkUtils.getCookie(request, SAMLSSOConstants.SAML_SSO_TOKEN_ID_COOKIE);
                 if (cookie != null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Found SamlssotokenId in the request object.");
+                    }
                     samlssoTokenId = cookie.getValue();
                 }
+        }
+        if (StringUtils.isEmpty(samlssoTokenId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("SamlssoTokenId is not found in the request object. Hence getting it from the context");
```

Comment:
Shall we change this like "SAMLSSOTokenId". Please check in other places as well.

Reply:
it is a cookie name